lib/OrdinaryDiffEqDifferentiation/src/derivative_wrappers.jl

```diff
@@ -75,198 +75,173 @@ function Base.showerror(io::IO, e::FirstAutodiffJacError)
     Base.showerror(io, e.e)
 end
 
-function derivative!(df::AbstractArray{<:Number}, f,
-        x::Union{Number, AbstractArray{<:Number}}, fx::AbstractArray{<:Number},
-        integrator, grad_config)
+function jacobian(f, x::AbstractArray{<:Number}, integrator)
     alg = unwrap_alg(integrator, true)
-    tmp = length(x) # We calculate derivative for all elements in gradient
-    autodiff_alg = alg_autodiff(alg)
-    if autodiff_alg isa AutoForwardDiff
-        T = if standardtag(alg)
-            typeof(ForwardDiff.Tag(OrdinaryDiffEqTag(), eltype(df)))
-        else
-            typeof(ForwardDiff.Tag(f, eltype(df)))
-        end
+    
+    # Update stats.nf
 
-        xdual = Dual{T, eltype(df), 1}(convert(eltype(df), x),
-            ForwardDiff.Partials((one(eltype(df)),)))
+    dense = alg_autodiff(alg) isa AutoSparse ? ADTypes.dense_ad(alg_autodiff(alg)) : alg_autodiff(alg)
 
-        if integrator.iter == 1
-            try
-                f(grad_config, xdual)
-            catch e
-                throw(FirstAutodiffTgradError(e))
-            end
-        else
-            f(grad_config, xdual)
-        end
+    if dense isa AutoForwardDiff
+        sparsity, colorvec = sparsity_colorvec(integrator.f, x)
+        maxcolor = maximum(colorvec)
+        chunk_size = (get_chunksize(alg) == Val(0) || get_chunksize(alg) == Val(nothing) ) ? nothing : get_chunksize(alg)
+        num_of_chunks = chunk_size === nothing ?
+                Int(ceil(maxcolor / getsize(ForwardDiff.pickchunksize(maxcolor)))) :
+                Int(ceil(maxcolor / _unwrap_val(chunk_size)))
```

Comment:
```suggestion
        num_of_chunks =  div(maxcolor, isnothing(chunk_size) ?
                getsize(ForwardDiff.pickchunksize(maxcolor)))) : _unwrap_val(chunk_size),
                RoundUp)
```

Reply:
Here you're implicitly assuming how the chunk size selection works in DI. Indeed I pick the chunk size based on the number of colors:

https://github.com/JuliaDiff/DifferentiationInterface.jl/blob/e541190e8e0a3193897fdc3fa42ae7ddf68e59bb/DifferentiationInterface/ext/DifferentiationInterfaceSparseMatrixColoringsExt/jacobian.jl#L75-L80

https://github.com/JuliaDiff/DifferentiationInterface.jl/blob/e541190e8e0a3193897fdc3fa42ae7ddf68e59bb/DifferentiationInterface/ext/DifferentiationInterfaceForwardDiffExt/utils.jl#L1-L8

But just so you know, it is not necessarily guaranteed by the API, and it is wrong if we switch to reverse- or mixed-mode Jacobians in the near future.